modules/animation/src/animationcontroller.cpp

```diff
@@ -469,46 +303,88 @@ void AnimationController::tick() {
+    const int numFrames =
+        std::max(2, static_cast<int>((lastTime - firstTime) / Seconds{1.0 / renderFPS.get()}));
+
+    // digits of the frame counter
+    const int digits = [&]() {
+        int d = 0;
+        int number(numFrames - 1);
+        while (number) {
+            number /= 10;
+            d++;
+        }
+        // use at least 4 digits, so we nicely overwrite the files from a previous test rendering
+        // with less frames
+        return std::max(d, 4);
+    }();
+
+    // Get all active canvases
+    std::vector<std::pair<Exporter*, std::string>> exporters;
+    network->forEachProcessor([&](Processor* p) {
+        if (p->isSink()) {
+            if (auto exporter = dynamic_cast<Exporter*>(p)) {
+                auto base = renderBaseName.get();
+                replaceInString(base, "UPN", p->getIdentifier());
+                exporters.emplace_back(exporter, base);
+            }
+        }
+    });
+
+    // Switch Buttons
+    renderAction.setReadOnly(true);
+    renderActionStop.setReadOnly(false);
+
+    util::OnScopeExit reset{[&]() {
         setState(AnimationState::Paused);
-        return;
-    }
+        renderAction.setReadOnly(false);
+        renderActionStop.setReadOnly(true);
+    }};
+
+    // render frames
+    for (int currentFrame = 0; currentFrame < numFrames; ++currentFrame) {
+        // Evaluate animation
+        Seconds newTime = firstTime + (lastTime - firstTime) / (numFrames - 1) * currentFrame;
+        eval(currentTime_, newTime);
+
+        lock.reset();
+
+        while (app_->getProcessorNetwork()->runningBackgroundJobs() > 0) {
+            app_->processFront();
+            app_->processEvents();
+            std::this_thread::yield();
+        }
+        app_->processEvents();
+
+        for (auto&& [exporter, base] : exporters) {
+            const auto file = fmt::format("{}{:0{}}", base, currentFrame, digits);
+            exporter->exportFile(renderLocation.get(), file, {writer.get()}, Overwrite::Yes);
+        }
+
+        lock.emplace(network);
 
-    // We render with equidistant steps
-    Seconds newTime = renderState_.firstTime;
-    if (renderState_.currentFrame >= 0) {
-        const double progress =
-            double(renderState_.currentFrame) / double(renderState_.numFrames - 1);
-        newTime += progress * (renderState_.lastTime - renderState_.firstTime);
+        if (state_ != AnimationState::Rendering) break;
     }
 
-    // Evaluate animation
-    eval(currentTime_, newTime);
+    using duration_double = std::chrono::duration<double, std::ratio<1>>;
+    auto seconds = std::chrono::duration_cast<duration_double>(
+                       std::chrono::high_resolution_clock::now() - start)
+                       .count();
+    LogInfo(fmt::format("Rendered {} frames in {:.3f} seconds, {:.3f} per frame", numFrames,
+                        seconds, seconds / numFrames));
```

Comment:
Does use of int not produce a warning?
```suggestion
                        seconds, seconds / static_cast<double>(numFrames)));
```

Reply:
I think double / int should be fine